### PR TITLE
Do not allow hiding income category group

### LINKED ISSUE #2579: [Bug]: Income category group has a hide option
### Verified issue does not already exist?

- [X] I have searched and found no existing issue
- [X] I will be providing steps how to reproduce the bug (in most cases this will also mean uploading a demo budget file)

### What happened?

After hiding out the entire income group, nothing visually changes in the GUI (the group name doesn't fade out) nor gives any visual signal to the user. 

This can lead to confusion as it effectively disables the ability of adding any income transaction throughout the application in case the "hide" option is selected on the entire Income group unintentionally. 

As per Discord discussion this should be probably fixed by removing the "Hide" option on the Income category group.

Category

### Where are you hosting Actual?

Docker

### What browsers are you seeing the problem on?

Chrome

### Operating System

Windows 11

### PR DESCRIPTION
**Pull request by psybers**
Closes #2579.

### CODE REVIEW

**Comment by MatissJanis** [comment]
Thanks, I think this makes sense. But would you mind creating a migration too to patch this category if it's already hidden?

(I'm on vacation, but will do a proper review when back on the weekend)

**Comment by psybers** [comment]
@MatissJanis Sure.  I don't think we need to check 'if' it is hidden, so I just set all income groups to not hidden.

File: packages/loot-core/migrations/1712784523000_unhide_input_group.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+BEGIN TRANSACTION;
+
+UPDATE category_groups
+SET
+  hidden = 0
+WHERE is_income;
```
Comment (joel-jeremy) [inline packages/loot-core/migrations/1712784523000_unhide_input_group.sql:6]: Should this be is_income = 1?

Comment (psybers) [reply packages/loot-core/migrations/1712784523000_unhide_input_group.sql:6]: Wouldn't hurt. Though I tested it (with a hidden non-income category) and it seemed to work as expected this way.

**Review by joel-jeremy — APPROVED** [review]
LGTM!

**Pull request merged**